### PR TITLE
[FLINK-13206][sql client]replace `use database xxx` with `use xxx` in sql client parser

### PR DESCRIPTION
**Pull request by zjuwangg**
## What is the purpose of the change

Replace `use database xxx` with `use xxx` in sql client parser 

## Brief change log
  - *modify use database in sql client*

## Verifying this change
This change is already covered by existing tests, such as *SqlCommandParserTest*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): ( no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: ( no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no )
  - The S3 file system connector: (no )

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable )

### CODE REVIEW

**Comment by zjuwangg** [comment]
cc @bowenli86 @xuefuz @lirui-apache to review.

**Comment by bowenli86** [comment]
please also update the corresponding documentation in `sql.md`

**Comment by zjuwangg** [comment]
> I'm thinking maybe we can keep 'USE DATABASE' to match 'USE CATALOG', and make both 'USE DATABASE XXX' and 'USE XXX' be interpreted as using a database?
> 
> Besides, can we add some negative tests?

I like the idea that keeps 'USE DATABASE' to match 'USE CATALOG', maybe we should think it furthermore in the DDL/DML design and support in sql parser.

**Comment by zjuwangg** [comment]
address the coments.cc @bowenli86

**Comment by bowenli86** [comment]
just ran a search and found `catalog.md` also need modification:

```
$ ag 'use database'
docs/dev/table/sql.md
325:USE DATABASE mydatabase;

docs/dev/table/catalog.md
66:Users can set current catalog and database via `TableEnvironment.useCatalog(...)` and `TableEnvironment.useDatabase(...)` in Table API, or `USE CATALOG ...` and `USE DATABASE ...` in Flink SQL.
340:Flink SQL> use database myDb;
```

**Comment by xuefuz** [comment]
I'm not convinced that we should introduce "use database database_name" grammar. I don't see a need, and keeping it simple will probably make our life easier.

**Comment by zjuwangg** [comment]
> just ran a search and found `catalog.md` also need modification:
> 
> ```
> $ ag 'use database'
> docs/dev/table/sql.md
> 325:USE DATABASE mydatabase;
> 
> docs/dev/table/catalog.md
> 66:Users can set current catalog and database via `TableEnvironment.useCatalog(...)` and `TableEnvironment.useDatabase(...)` in Table API, or `USE CATALOG ...` and `USE DATABASE ...` in Flink SQL.
> 340:Flink SQL> use database myDb;
> ```

Sorry to miss these, I have correct them in the lastest commit.

**Comment by zjuwangg** [comment]
cc @bowenli86 to review.

**Comment by bowenli86** [comment]
LGTM, will merge once CI passes

**Comment by bowenli86** [comment]
can you rerun the tests. seems to have some failures

**Comment by zjuwangg** [comment]
> can you rerun the tests. seems to have some failures

It's caused by the python module, I just rebase the master.

**Comment by zjuwangg** [comment]
@bowenli86 I have tested in my traivs account, there is a failure in blink-planner that is not related to this commit.
[CI BUILD](https://travis-ci.com/zjuwangg/flink/jobs/217144849)

File: docs/dev/table/catalog.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ select * from mydb2.myTable2
 
 `CatalogManager` always has a built-in `GenericInMemoryCatalog` named `default_catalog`, which has a built-in default database named `default_database`. If no other catalog and database are explicitly set, they will be the current catalog and current database by default. All temp meta-objects, such as those defined by `TableEnvironment#registerTable`  are registered to this catalog. 
 
-Users can set current catalog and database via `TableEnvironment.useCatalog(...)` and `TableEnvironment.useDatabase(...)` in Table API, or `USE CATALOG ...` and `USE DATABASE ...` in Flink SQL.
+Users can set current catalog and database via `TableEnvironment.useCatalog(...)` and `TableEnvironment.useDatabase(...)` in Table API, or `USE CATALOG ...` and `USE ...` in Flink SQL.
```
[the remark this answers is not in the document]
Comment (zjuwangg) [reply docs/dev/table/catalog.md]: @twalthr Nice scrupulous suggestion!

**Comment by zjuwangg** [comment]
cc @bowenli86 again after addressing twalthr's comments.

**Pull request closed without merging**